sphinxcontrib/spelling/__init__.py

```diff
@@ -54,6 +54,8 @@ def setup(app):
     app.add_config_value('spelling_filters', [], 'env')
     # Set a user-provided list of files to ignore
     app.add_config_value('spelling_exclude_patterns', [], 'env')
+    # Set valid word characters
+    app.add_config_value('spelling_valid_word_characters', 'en_US', 'env')
```

Comment:
I think we want the default value here to be a string with the extra characters, right? So it would include an apostrophe and hyphen, for example. Or perhaps be empty, and the help text would explain to add those characters if desired?

Reply:
Oh, derp. Copy-pasta sloppiness.

Actually when I did my test, the values were hard-coded in a tuple:

```
            self._valid_chars = ("'","-",)
```

So I am not sure whether I should pass in a tuple of strings or a string and split it into a tuple of strings. What do you think?

Reply:
A list or a string makes sense as a sequence of a common type.

What does the pyenchant code expect? What's the default?

Reply:
A tuple of strings.

Reply:
See:

https://github.com/pyenchant/pyenchant/blob/main/enchant/tokenize/en.py#L88-L100

Reply:
OK, that's an interesting choice. I guess to make it immutable?

If that's the type expected, we can either use that or use something we can convert to that before passing the value to PyEnchant. A single string acting as a set of characters makes some sense, but since quotes are going to be commonly added maybe using 1 string makes it a bit ugly to get the syntax right. A list would be more semantically correct.

I guess I don't have a strong opinion, though. What makes sense to you?

Reply:
I went with a tuple for now.

If I change it to a list, would I make the conversion from a list to a tuple at this point?

https://github.com/sphinx-contrib/spelling/blob/4609309b73c60318766bf9336569068e4337f24f/sphinxcontrib/spelling/checker.py#L37

Reply:
That looks like the right place for a conversion. Let's not get hung up on the semantics of list vs. tuple -- if the code expects a tuple, we can go with tuple.

Reply:
Sounds good.